core/src/main/java/org/epcc/ps/core/algorithm/implement/CoreAlgorithmImp.java

```diff
@@ -0,0 +1,42 @@
+package org.epcc.ps.core.algorithm.implement;  
+
+import org.epcc.ps.core.algorithm.CoreAlgorithm;
+
+/**
+ * @author jiahao.cao
+ * Created on 11/10/2017
+ */
+public class CoreAlgorithmImp implements CoreAlgorithm{
+	/***
+	 * Get the new number of Hares in this grid
+	 * 
+	 * */
+	@Override
+	public double getHaresNum(double hareOldNum, double hareOldLeft,
+		   double hareOldRight,double hareOldAbove,double hareOldFollowing, 
+		   double hareBirthRate, double pumaPredationRate, double pumaOldNum, 
+		   double hareDifussionRate, double time, int dryNum){
+		
+			double hareNewNum;
+			hareNewNum = hareOldNum + time * ( hareBirthRate * hareOldNum - pumaPredationRate * hareOldNum * pumaOldNum + hareDifussionRate * ((hareOldLeft + hareOldRight + hareOldAbove + hareOldFollowing) - dryNum * hareOldNum));
```

Comment:
line separation for code style ( code format )
donot put all the arguments in a single line